encoding/thrift/inbound.go

```diff
@@ -101,58 +72,89 @@ func (t thriftUnaryHandler) Handle(ctx context.Context, treq *transport.Request,
 	}
 
 	if err := call.WriteToResponse(rw); err != nil {
+		// not reachable
 		return err
 	}
 
-	err = proto.EncodeEnveloped(wire.Envelope{
-		Name:  res.Body.MethodName(),
-		Type:  res.Body.EnvelopeType(),
-		SeqID: envelope.SeqID,
-		Value: value,
-	}, rw)
+	err = responder.EncodeResponse(value, wire.Reply, rw)
 	if err != nil {
 		return errors.ResponseBodyEncodeError(treq, err)
 	}
 
 	return nil
 }
 
-// TODO(apb): reduce commonality between Handle and HandleOneway
-
 func (t thriftOnewayHandler) HandleOneway(ctx context.Context, treq *transport.Request) error {
-	if err := errors.ExpectEncodings(treq, Encoding); err != nil {
+	ctx, _, reqValue, _, err := handleRequest(ctx, treq, wire.OneWay, t.Protocol, t.Enveloping)
+	if err != nil {
 		return err
 	}
 
-	ctx, call := encodingapi.NewInboundCall(ctx)
-	if err := call.ReadFromRequest(treq); err != nil {
-		return err
-	}
+	return t.OnewayHandler(ctx, reqValue)
+}
 
-	buf := bufferpool.Get()
-	defer bufferpool.Put(buf)
-	if _, err := buf.ReadFrom(treq.Body); err != nil {
-		return err
+// handleRequest is a utility shared by Unary and Oneway handlers, to decode the request, regardless of enveloping
+func handleRequest(
+	parentCtx context.Context,
+	treq *transport.Request,
+	reqEnvelopeType wire.EnvelopeType,
+	proto protocol.Protocol,
+	enveloping bool,
+) (
+	ctx context.Context,
+	call *encodingapi.InboundCall,
+	reqValue wire.Value,
+	responder protocol.Responder,
```

Comment:
Please document the return values in the function documentation or next to the return values.

It would be useful to know which values are meaningless in what contexts.